users/views.py

```diff
@@ -1,10 +1,57 @@
-from django.contrib.auth.forms import UserCreationForm
-from django.urls import reverse_lazy
-from django.views import generic
+# from django.contrib.auth.forms import UserCreationForm
+# from django.urls import reverse_lazy
+# from django.views import generic
+from rest_framework import viewsets, generics
+from .models import UserProfile
+from .serializers import UserSerializer, CreateUserSerializer, CustomAuthTokenSerializer
+from users.permisions import IsOwnerIsAdmin, IsOwner
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.authtoken.views import ObtainAuthToken
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
 
 
-class SignUp(generic.CreateView):
-    form_class = UserCreationForm
-    success_url = reverse_lazy('login')
-    template_name = 'registration/signup.html'
+class UserViewSet(viewsets.ModelViewSet):
+    queryset = UserProfile.objects.all()
+    serializer_class = UserSerializer
+    permission_classes = (IsAuthenticated, IsOwnerIsAdmin)
+
+
+class UsersView(generics.CreateAPIView):
+    serializer_class = CreateUserSerializer
+    permission_classes = (IsAuthenticated, IsOwnerIsAdmin)
+
+
+class CustomAuthToken(ObtainAuthToken):
+
+    def post(self, request, *args, **kwargs):
+        serializer = CustomAuthTokenSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.validated_data['user']
+        token, created = Token.objects.get_or_create(user=user)
+        return Response({
+            'token': token.key,
+            'username': user.username,
+        })
+
+# class SignUp(generic.CreateView):
+#     form_class = UserCreationForm
+#     success_url = reverse_lazy('login')
+#     template_name = 'registration/signup.html'
+
+
+class UserDetail(generics.RetrieveAPIView):
+    queryset = UserProfile.objects.all()
+    serializer_class = UserSerializer
```

Comment:
Currently, `queryset` and `serializer_class` are not used. You get the user without using `queryset` and you create response with creating a dictionary. 

Can you please read generic view examples from the documentation ? We should not update `get` method. Please check `get_object` method.